File: php/hamming/hamming.php
<?php
function distance($a, $b)
{
    if(strlen($a) != strlen($b)){
        throw new InvalidArgumentException('DNA strands must be of equal length.');
    }

    $m = 0;
    foreach(str_split($a) as $k => $c){
        if($b[$k] != $c){
            $m++;
        }
    }
    return $m;

}

File: ruby/hamming/hamming.rb
# Calculate the Hamming difference between two DNA strands
class Hamming
  def self.compute(old_strand, new_strand)
    validate(old_strand, new_strand)

    old_strand
      .chars
      .zip(new_strand.chars)
      .count { |oc, nc| oc != nc }
  end

  def self.validate(old_strand, new_strand)
    return if old_strand.size == new_strand.size

    raise ArgumentError, 'Must be same size'
  end
end
